provider/cmd/crd2pulumi/gen/generate.go

```diff
@@ -130,87 +128,86 @@ func (gen *CustomResourceGenerator) Name() string {
 	return gen.Plural + "." + gen.Group
 }
 
-// GetVersionNames returns a slice of the versions supported by this CRD.
-func (gen *CustomResourceGenerator) GetVersionNames() []string {
+// VersionKeys returns a slice of the versions supported by this CRD.
+func (gen *CustomResourceGenerator) VersionKeys() []string {
 	versionNames := make([]string, 0, len(gen.Versions))
 	for versionName := range gen.Versions {
 		versionNames = append(versionNames, versionName)
 	}
 	return versionNames
 }
 
-func (gen *CustomResourceGenerator) GenerateCode() error {
-	objectTypeSpecs := GetObjectTypeSpecs(gen.Versions, gen.Name(), gen.Kind)
+// VersionNames returns a slice of the full names of each version, in the format
+// <plural>.<group>/<version>.
+func (gen *CustomResourceGenerator) VersionNames() []string {
+	versions := gen.VersionKeys()
+	name := gen.Name()
+	for i, version := range versions {
+		versions[i] = name + "/" + version
+	}
+	return versions
+}
+
+// getVersion returns the <version> field of a string in the format
+// <plural>.<group>/<version>
+func getVersion(versionName string) string {
+	return strings.Split(versionName, "/")[1]
```

Comment:
Is this guaranteed to include a `/`? If not, this line can panic.

Might be worth including an assertion about the number of elements prior to accessing, so it at least prints a nice error message if this invariant is violated.